functions/go/starlark/README.md

```diff
@@ -8,34 +8,46 @@ Run a Starlark script to mutate or validate resources.
 
 <!--mdtogo-->
 
-### FunctionConfig
+[Starlark] is a dialect of Python. It is commonly used as a configuration
+language. It is an untyped dynamic language with high-level data types,
+first-class functions with lexical scope, and garbage collection. You can find
+the spec for the Starlark language [here][spec]. You can also find its API
+reference [here][apiref].
 
 <!--mdtogo:Long-->
 
-[Starlark] is a dialect of Python. It is commonly used as a configuration language. It is an untyped dynamic language with
-high-level data types, first-class functions with lexical scope, and garbage
-collection.
+### Function
+
+[starlark] function contains an interpreter for the Starlark language. You can
+provide a Starlark script in the `functionConfig` to mutate or validate your
+resources. It is a light-weight way to write a KRM function. It is recommended
+to only build simple function with it. Generally, if you Starlark script is
+longer than 20 lines, you may want to consider build a function with our
+[Golang SDK] or [TypeScript SDK].
+
+### FunctionConfig
 
-The starlark function currently uses `StarlarkRun` as the function config. It
-looks like this:
+The starlark function accepts a CRD of kind `StarlarkRun` as the
+`functionConfig`. It looks like this:
 
 ```yaml
 apiVersion: fn.kpt.dev/v1alpha1
 kind: StarlarkRun
 metadata:
   name: my-star-fn
 source: |
-  # set the namespace on each resource
-  def run(resources, ns_value):
-    for resource in resources:
-    # mutate the resource
-    resource["metadata"]["namespace"] = ns_value
-  run(ctx.resource_list["items"], "prod")
+  # Starlark source lives here.
+  # You can mutate or validate the resources in ctx.resource_list["items"].
```

Comment:
There's lot more to know here about how to write such a function. For example:

- you need a function to call
- you need to understand the the Functions Spec: https://kpt.dev/book/05-developing-functions/01-functions-specification